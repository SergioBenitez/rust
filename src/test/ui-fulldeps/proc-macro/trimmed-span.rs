// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

// aux-build:trimmed-span.rs
// ignore-stage1

#![feature(proc_macro_non_items)]

extern crate trimmed_span;

use trimmed_span::trimmed;

fn main() {
    // This one emits no error.
    trimmed!("");

    // Exactly one 'hi'.
    trimmed!("hi"); //~ ERROR found 'hi's

    // Now two, back to back.
    trimmed!("hihi"); //~ ERROR found 'hi's

    // Now three, back to back.
    trimmed!("hihihi"); //~ ERROR found 'hi's

    // Now several, with spacing.
    trimmed!("why I hide? hi!"); //~ ERROR found 'hi's
    trimmed!("hey, hi, hidy, hidy, hi hi"); //~ ERROR found 'hi's
    trimmed!("this is a hi, and this is another hi"); //~ ERROR found 'hi's
    trimmed!("how are you this evening"); //~ ERROR found 'hi's
    trimmed!("this is highly eradic"); //~ ERROR found 'hi's
}
