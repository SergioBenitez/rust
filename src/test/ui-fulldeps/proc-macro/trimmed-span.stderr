error: found 'hi's
  --> $DIR/trimmed-span.rs:25:5
   |
LL |     trimmed!("hi"); //~ ERROR found 'hi's
   |     ^^^^^^^^^^^^^^^
   |
note: here
  --> $DIR/trimmed-span.rs:25:15
   |
LL |     trimmed!("hi"); //~ ERROR found 'hi's
   |               ^^

error: found 'hi's
  --> $DIR/trimmed-span.rs:28:5
   |
LL |     trimmed!("hihi"); //~ ERROR found 'hi's
   |     ^^^^^^^^^^^^^^^^^
   |
note: here
  --> $DIR/trimmed-span.rs:28:15
   |
LL |     trimmed!("hihi"); //~ ERROR found 'hi's
   |               ^^^^

error: found 'hi's
  --> $DIR/trimmed-span.rs:31:5
   |
LL |     trimmed!("hihihi"); //~ ERROR found 'hi's
   |     ^^^^^^^^^^^^^^^^^^^
   |
note: here
  --> $DIR/trimmed-span.rs:31:15
   |
LL |     trimmed!("hihihi"); //~ ERROR found 'hi's
   |               ^^^^^^

error: found 'hi's
  --> $DIR/trimmed-span.rs:34:5
   |
LL |     trimmed!("why I hide? hi!"); //~ ERROR found 'hi's
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: here
  --> $DIR/trimmed-span.rs:34:21
   |
LL |     trimmed!("why I hide? hi!"); //~ ERROR found 'hi's
   |                     ^^    ^^

error: found 'hi's
  --> $DIR/trimmed-span.rs:35:5
   |
LL |     trimmed!("hey, hi, hidy, hidy, hi hi"); //~ ERROR found 'hi's
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: here
  --> $DIR/trimmed-span.rs:35:20
   |
LL |     trimmed!("hey, hi, hidy, hidy, hi hi"); //~ ERROR found 'hi's
   |                    ^^  ^^    ^^    ^^ ^^

error: found 'hi's
  --> $DIR/trimmed-span.rs:36:5
   |
LL |     trimmed!("this is a hi, and this is another hi"); //~ ERROR found 'hi's
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: here
  --> $DIR/trimmed-span.rs:36:16
   |
LL |     trimmed!("this is a hi, and this is another hi"); //~ ERROR found 'hi's
   |                ^^       ^^       ^^             ^^

error: found 'hi's
  --> $DIR/trimmed-span.rs:37:5
   |
LL |     trimmed!("how are you this evening"); //~ ERROR found 'hi's
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: here
  --> $DIR/trimmed-span.rs:37:28
   |
LL |     trimmed!("how are you this evening"); //~ ERROR found 'hi's
   |                            ^^

error: found 'hi's
  --> $DIR/trimmed-span.rs:38:5
   |
LL |     trimmed!("this is highly eradic"); //~ ERROR found 'hi's
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: here
  --> $DIR/trimmed-span.rs:38:16
   |
LL |     trimmed!("this is highly eradic"); //~ ERROR found 'hi's
   |                ^^     ^^

error: aborting due to 8 previous errors

